source/initial_topography_model/interface.cc

```diff
@@ -0,0 +1,173 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/global.h>
+#include <aspect/initial_topography_model/interface.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/std_cxx11/tuple.h>
+#include <list>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    Interface<dim>::~Interface ()
+    {}
+
+
+    template <int dim>
+    void
+    Interface<dim>::initialize ()
+    {}
+
+    template <int dim>
+    void
+    Interface<dim>::
+    declare_parameters (dealii::ParameterHandler &)
+    {}
+
+
+    template <int dim>
+    void
+    Interface<dim>::parse_parameters (dealii::ParameterHandler &)
+    {}
+
+
+// -------------------------------- Deal with registering initial topography models and automating
+// -------------------------------- their setup and selection at run time
+
+    namespace
+    {
+      std_cxx11::tuple
+      <void *,
+      void *,
+      internal::Plugins::PluginList<Interface<2> >,
+      internal::Plugins::PluginList<Interface<3> > > registered_plugins;
+    }
+
+
+
+    template <int dim>
+    void
+    register_initial_topography_model (const std::string &name,
+                                       const std::string &description,
+                                       void (*declare_parameters_function) (ParameterHandler &),
+                                       Interface<dim> *(*factory_function) ())
+    {
+      std_cxx11::get<dim>(registered_plugins).register_plugin (name,
+                                                               description,
+                                                               declare_parameters_function,
+                                                               factory_function);
+    }
+
+
+    template <int dim>
+    Interface<dim> *
+    create_initial_topography_model (ParameterHandler &prm)
+    {
+      std::string model_name;
+      prm.enter_subsection ("Initial topography model");
```

Comment:
Should this be a section inside `section Geometry model`? I think that would nicely group where it belongs.